pkg/cmd/experimental/syncgroups/config/types.go

```diff
@@ -0,0 +1,204 @@
+package config
+
+import (
+	ldaputil "github.com/openshift/origin/pkg/auth/authenticator/password/ldappassword/util"
+	"github.com/openshift/origin/pkg/client"
+)
+
+// LDAPSchemaType determines which type of LDAP server schema is present for the sync job.
+type LDAPSchemaType string
+
+const (
+	// LDAPSyncTypeFirstClassGroups defines an LDAP schema where groups are stored as first-class LDAP
+	// entries with an attribute containing a list of members, either with members listed with their
+	// DN in the attribute members listed with some other UID in the attribute.
+	LDAPSchemaTypeFirstClassGroups LDAPSchemaType = "FirstClassGroups"
+
+	// LDAPSyncTypeFirstClassUserss defines an LDAP schema where users are stored as first-class LDAP
+	// entries with an attribute containing a list of groups they are a member of, with groups listed
+	// by a unique name, with no additional group metadata in the LDAP server.
+	LDAPSchemaTypeFirstClassUserss LDAPSchemaType = "FirstClassUserss"
+
+	// LDAPSyncTypeFirstClassGroupsAndUsers defines an LDAP schema where users are stored as first-class
+	// LDAP entries with an attribute containing a list of groups they are a member of, either with
+	// groups listed by DN, with the first-class group entries holding additional metadata groups listed
+	// by some other UID, with the first-class group entries holding additional metadata.
+	LDAPSchemaTypeFirstClassGroupsAndUsers LDAPSchemaType = "FirstClassGroupsAndUsers"
+)
+
+// GroupSyncScope determines the scope of the group sync operation
+type GroupSyncScope string
+
+const (
+	// GroupSyncScopeAll determines that all groups from the source should be synced
+	GroupSyncScopeAll GroupSyncScope = "All"
+	// GroupSyncScopeWhitelist determines that a whitelist of groups from the source should be synced
+	GroupSyncScopeWhitelist GroupSyncScope = "Whitelist"
+)
+
+// GroupSyncSource determines the source of the groups to be synced
+type GroupSyncSource string
+
+const (
+	// GroupSyncSourceLDAP determines that the groups to be synced are determined from an LDAP record
+	GroupSyncSourceLDAP GroupSyncSource = "LDAP"
+	// GroupSyncSourceOpenShift determines that the groups to be synced are determined from OpenShift records
+	GroupSyncSourceOpenShift GroupSyncSource = "OpenShift"
+)
+
+// LDAPSyncConfig contains all of the configuration information for an LDAP group sync job.
+type LDAPSyncConfig struct {
```

Comment:
Is the intent of this object to be serialized or built from a serialized config?